### PR TITLE
GH-1158 Fixes Surface button annotations for FF

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-1158

@mlzummo @dtsnell4 @GordonSmith Please review

http://rawgit.com/jbrundage/Visualization/GH-1158-FF-click/demos/dermatology.html?src/layout/Surface

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
fix for common/Surface too

**Comment by mlzummo** [comment]
and any other place that might need it

File: src/layout/Surface.js [hunk 1]
```diff
@@ -85,7 +85,7 @@
         surfaceButtons.enter().append("button").classed("surface-button",true)
             .each(function (button, idx) {
                 var el = context._surfaceButtons[idx] = d3.select(this)
-                    .attr("class", "surface-button " + button.class)
+                    .classed(button.class,button.class !== undefined)
```
Comment (GordonSmith) [inline src/layout/Surface.js]: Move into an if statement - I don't like the ```.classed(undefind, false)``` case.

Comment (jbrundage) [inline src/layout/Surface.js]: Will do, but out of curiosity, could this cause any problems?

Comment (GordonSmith) [inline src/layout/Surface.js]: I don't know for sure and suspect its probably ok - but I suspect it is passed to the DOMNode which is probably wasteful at best.

File: src/layout/Surface.js [hunk 2]
```diff
@@ -85,18 +85,20 @@
         surfaceButtons.enter().append("button").classed("surface-button",true)
             .each(function (button, idx) {
                 var el = context._surfaceButtons[idx] = d3.select(this)
-                    .attr("class", "surface-button " + button.class)
```
Comment (GordonSmith) [inline src/layout/Surface.js]: This line is till needed?

**Comment by mlzummo** [comment]
can we close this in favor of new PR

**Comment by GordonSmith** [comment]
Sure

**Comment by mlzummo** [comment]
@GordonSmith can we close this PR

**Comment by GordonSmith** [comment]
Closed in favor of:  https://github.com/hpcc-systems/Visualization/pull/1228

**Pull request closed without merging**